src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2421,30 +2505,36 @@ public override BoundNode VisitAddressOfOperator(BoundAddressOfOperator node)
             return null;
         }
 
-        private void WarnOnNullReferenceArgument(BoundExpression argument, TypeSymbolWithAnnotations argumentType, ParameterSymbol parameter, bool expanded)
+        private bool ReportNullReferenceArgumentIfNecessary(BoundExpression argument, TypeSymbolWithAnnotations argumentType, ParameterSymbol parameter, TypeSymbolWithAnnotations paramType)
         {
-            var paramType = parameter.Type;
-
             if (argumentType?.IsNullable == true)
             {
-                if (expanded)
-                {
-                    paramType = ((ArrayTypeSymbol)parameter.Type.TypeSymbol).ElementType;
-                }
-
-                if (paramType.IsReferenceType && paramType.IsNullable == false && !CheckNullAsNonNullableReference(argument))
+                if (paramType.IsReferenceType && paramType.IsNullable == false)
                 {
-                    ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullReferenceArgument, argument.Syntax,
-                        new FormattedSymbol(parameter, SymbolDisplayFormat.ShortFormat),
-                        new FormattedSymbol(parameter.ContainingSymbol, SymbolDisplayFormat.MinimallyQualifiedFormat));
+                    if (!CheckNullAsNonNullableReference(argument))
+                    {
+                        ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullReferenceArgument, argument.Syntax,
+                            new FormattedSymbol(parameter, SymbolDisplayFormat.ShortFormat),
+                            new FormattedSymbol(parameter.ContainingSymbol, SymbolDisplayFormat.MinimallyQualifiedFormat));
+                    }
+                    return true;
                 }
             }
+            return false;
+        }
+
+        private void ReportArgumentWarnings(BoundExpression argument, TypeSymbolWithAnnotations argumentType, ParameterSymbol parameter)
+        {
+            Debug.Assert(!parameter.IsParams);
```

Comment:
>!parameter.IsParams [](start = 25, length = 19)

This looks like a wrong assumption

Reply:
Removed and added test.

---
In reply to: [181206249](https://github.com/dotnet/roslyn/pull/26107#discussion_r181206249) [](ancestors = 181206249)